torch/csrc/jit/runtime/static/memory_planner.h

```diff
@@ -199,8 +192,25 @@ class MemoryPlanner {
   // We don't have any guarantee that the model doesn't change the
   // Storage for managed tensors out from under us during execution,
   // so we have to check the StorageImpls each time we deallocate.
-  std::vector<std::pair<size_t, at::StorageImpl>>
-      managed_tensor_storage_impls_{};
+
+  // We will use placement-new to add new storages to this buffer of bytes.
+  alignas(at::StorageImpl) unsigned char* storages_buffer_;
+
+  // This pointer just points to the storage buffer with the proper type
+  at::StorageImpl* storages_;
+
+  // Contains the size (in bytes) of the data to be allocated for each storage
+  size_t* storages_nbytes_;
+
+  // Current number of storages that have been placed into the storage buffer
+  size_t storages_size_;
+
+  // Total capacity of the storage buffer
+  size_t storages_capacity_;
```

Comment:
I'm not sure why you need all of these sizes; a simple count of how many storages are in the buffer feels sufficient

Reply:
Oh I see the old code stored a pair. Store the size_t in a normal vector lol